api/types/apitypes.go

```diff
@@ -460,6 +460,29 @@ func (s *Status) API() APIStatus {
 	}
 }
 
+// Happy describes just how happy dcrdata is.
+type Happy struct {
+	Happy           bool  `json:"happy"`
+	APIReady        bool  `json:"api_ready"`
+	TipAge          int64 `json:"tip_age"`
+	NodeConnections int64 `json:"node_connections"`
+}
+
+// Happy indicates how dcrdata is or isn't happy.
+func (s *Status) Happy() Happy {
+	s.RLock()
+	blockAge := time.Since(time.Unix(s.dbLastBlockTime, 0))
+	h := Happy{
+		APIReady:        s.ready,
+		TipAge:          int64(blockAge.Seconds()),
+		NodeConnections: s.nodeConnections,
+	}
+	s.RUnlock()
+
+	h.Happy = h.APIReady && blockAge < 90*time.Minute && h.NodeConnections > 0
```

Comment:
I'm unsure if 90 minutes is a safe happy threshold.  Will have to check the distribution to pick a duration that is statistically extremely unlikely.

Reply:
For an exponential distribution with a mean of 300 seconds, the probability of a block taking 5400 seconds to mine is about 0.9999999847700203 (close to 1 - 5e-8).  Network incidents not considered, it seems that 90 minutes is not too long.

Reply:
BTW, that's computed from `1 - exp(-5400/300) = 1 - 1/exp(18)`.

Reply:
I like those numbers.